stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -294,3 +294,36 @@ function eigvals!(A::StridedMatrix{T}, F::LU{T,<:StridedMatrix}; sortby::Union{F
     rdiv!(A, U)
     return eigvals!(A; sortby)
 end
+
+
+function eigen(A::Hermitian{Complex{T}, <:Tridiagonal}; kwargs...) where {T}
+    (; dl, d, du) = parent(A)
+    N = length(d)
+    if N <= 1
+        eigen(parent(A); kwargs...)
+    else
+        if A.uplo == 'U'
+            E = du'
+            Er = abs.(du)
+        else
+            E = dl
+            Er = abs.(E)
+        end
+        S = Vector{Complex{T}}(undef, N)
+        S[1] = 1
+        S[2] = Er[1] == 0 ? one(T) : S[1] * sign(E[i]])
```

Comment:
Why unfold this from the loop - that fails in the corner case `N == 1`.

Reply:
There's an `if N <= 1` line up above that deals with that. You're right that this doesn't really need to be unrolled though, I'll change it.